src/Illuminate/Database/Query/Builder.php

```diff
@@ -2264,6 +2264,27 @@ public function toSql()
         return $this->grammar->compileSelect($this);
     }
 
+    /**
+     * Get the SQL representation of the query replacing all bindings.
+     *
+     * @return string
+     */
+    public function toSqlWithBindings()
+    {
+        return Str::replaceArray(
+            '?',
+            collect($this->getBindings())
+                ->map(function ($i) {
+                    if (is_object($i)) {
+                        $i = (string) $i;
```

Comment:
This will throw a fatal if the object in question does not implement __toString(), is that desired or should it be catched and turned into a logic exception?

Reply:
I tested it with SQLite, SQL Server, MySQL and PostgreSQL, and the tests still passing (i'll add the tests as a separate commit and we can drop if later if it's unnecesssary). Can you provide an example where you think the object does not implement `__toString()`?

Reply:
Sure:

~~~php
User::query()->where('updated_at', '>', new DateTime('yesterday'))->get();
~~~

https://3v4l.org/Xi2hT

~~~
Fatal error: Uncaught Error: Object of class DateTime could not be converted to string
~~~

Reply:
Just to clarify:

The query from the first snippet above works on a stock Laravel installation.

The link was to show the `\DateTime` class fails to be cast to a string.